### PR TITLE
Exclude unsupported data types

### PR DESCRIPTION
**Pull request by naoyam**
bfloat16 fails on my system (Titan RTX):

```
CUDA NVRTC compile error: ptxas application ptx input, line 45; error   : Feature '.bf16' requires .target sm_80 or higher
ptxas application ptx input, line 45; error   : Feature 'cvt with .bf16' requires .target sm_80 or higher
ptxas fatal   : Ptx assembly aborted due to errors

Exception raised from nvrtcCompile at ../torch/csrc/jit/codegen/cuda/executor_utils.cpp:1154 (most recent call first):
```

This PR filters out unsupported types from the `zeros` test, etc.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/type.cpp [hunk 1]
```diff
@@ -160,6 +162,17 @@ DataType getTypeFromComplexType(DataType dtype) {
   }
 }
 
+bool isSupportedType(DataType dtype) {
+  auto prop = at::cuda::getCurrentDeviceProperties();
+  auto major_ver = prop->major;
+  switch (dtype) {
+    case DataType::BFloat16:
+      return major_ver >= 8;
+    default:
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/type.cpp:171]: Haven't checked other types. Probably we should also check `half`.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/type.cpp:171]: `half` seems to be supported since v 5.3, so should be fine.

Comment (zasdfgbnm) [reply torch/csrc/jit/codegen/cuda/type.cpp:171]: I think half should be supported starting from something like SM35, which is pretty old and no longer supported by PyTorch.

File: torch/csrc/jit/codegen/cuda/type.h [hunk 2]
```diff
@@ -101,6 +101,8 @@ int getVectorSizeFromType(DataType dtype);
 DataType getTypeFromVectorType(DataType dtype);
 // Return the corresponding scalar of a complex type
 DataType getTypeFromComplexType(DataType dtype);
+// Return if the datatype is supported on the current device
+TORCH_CUDA_CU_API bool isSupportedType(DataType dtype);
```
Comment (zasdfgbnm) [inline torch/csrc/jit/codegen/cuda/type.h]: nit: Would `isSupportedTypeByDevice` be a better name?

**Review by zasdfgbnm — APPROVED** [review]
Thanks a lot for fixing!

**Pull request merged**